### DISCUSSION TITLE
export by using terminal

### DISCUSSION THREAD

**Discussion by sangkookchoi** [retorquere/zotero-better-bibtex#2773] [Q&A] [answered]
Hi, Would it be possible to export the Zotero library from terminal?
I want  to  initiate the library export from terminal (probably using python) in the format of json file.
Is this possible by using Better BibTeX?

**Comment by retorquere** [comment]
Yes: https://retorque.re/zotero-better-bibtex/exporting/pull/index.html

**Comment by sangkookchoi** [comment, marked as answer]
Hi Thanks for your prompt reply.
Can you please give me a detailed instruction? It would be great if you can give me an example python script. I already read the instruction in the webpage but I don't know how to make a python script. Thanks for your help in advance.
With regards,
Sang.

**Reply by retorquere** [reply]
```
import urllib.request, json
lib = json.loads(urllib.request.urlopen('http://127.0.0.1:23119/better-bibtex/export/library?/1/library.jzon').read().decode())
```

if you want it in a string, or
```
import urllib.request
urllib.request.urlretrieve('http://127.0.0.1:23119/better-bibtex/export/library?/1/library.jzon', "library.json")
```

if you want it in a file.

But if you want general python support, you will have to ask on stack overflow.

**Comment by sangkookchoi** [comment]
Dear Emiliano, 

Thanks for your prompt reply. It works. I appreciate it. 

1) two more questions. When I export a collection with ~40 articles, it works fine with any format. However, when I try to export my whole library ("My Library") with ~4000 articles, it only works in JSON format. When I try with biblatex format it returns the following error: 

HTTPError: HTTP Error 500: Internal Server Error. 

Is this error expected? Is there any way to resolve this issue?

2) Would it be possible to export in the "better bibtex json" format from terminal?

Thanks in advance.

**Reply by retorquere** [reply]
> Is this error expected? 

It's not expected

> Is there any way to resolve this issue?

You'd have to open an issue so I can investigate

> Would it be possible to export in the "better bibtex json" format from terminal?

The code above gives you BetterBibTeX JSON.

**Comment by sangkookchoi** [comment]
I appreciate it:)
